src/helpers/steg/omdeg.ts

```diff
@@ -93,47 +101,92 @@ export const erPeriodeDatoerValgt = (periode: IPeriode) => {
+      perioderBoddIUtlandet?.some(
+        (utenlandsopphold) =>
+          utenlandsopphold.begrunnelse.verdi === '' ||
+          !utenlandsopphold.begrunnelse
+      );
+
+    return søkerBosattINorgeSisteTreÅr?.verdi === false
+      ? finnesUtenlandsperiodeUtenBegrunnelse
+        ? false
+        : true
+      : søkerBosattINorgeSisteTreÅr?.verdi
+      ? true
+      : false;
+  } else return false;
 };
 
-export const erStegFerdigUtfylt = (
-  sivilstatus: ISivilstatus,
-  medlemskap: IMedlemskap
+const erDatoSøktSeparasjonGyldig = (
+  datoSøktSeparasjon: IDatoFelt | undefined
 ): boolean => {
-  const { harSøktSeparasjon, datoSøktSeparasjon, datoFlyttetFraHverandre } =
-    sivilstatus;
-  const datoFlyttetfraHverandreErUtfyltOgGyldig =
-    datoFlyttetFraHverandre?.verdi &&
-    erDatoGyldigOgInnaforBegrensninger(
-      datoFlyttetFraHverandre?.verdi,
-      DatoBegrensning.AlleDatoer
-    );
-
-  const datoSøktSeparasjonerUtfyltOgGyldig =
+  return !!(
     datoSøktSeparasjon?.verdi &&
     erDatoGyldigOgInnaforBegrensninger(
       datoSøktSeparasjon?.verdi,
       DatoBegrensning.TidligereDatoer
+    )
+  );
+};
+
+const erSpørsmålOmUformeltGiftBesvart = (
+  sivilstatus: ISivilstatus
+): boolean => {
+  return sivilstatus.erUformeltGift?.verdi !== undefined;
+};
+
+const erSpørsmålOmUformeltSeparertEllerSkiltBesvart = (
+  sivilstatus: ISivilstatus
+): boolean => {
+  return sivilstatus.erUformeltSeparertEllerSkilt?.verdi !== undefined;
+};
+
+const erSpørsmålOmSøktSeparasjonUtfylt = (
+  sivilstatus: ISivilstatus
+): boolean => {
+  const { harSøktSeparasjon, datoSøktSeparasjon } = sivilstatus;
+
+  const datoSøktSeparasjonerUtfyltOgGyldig =
+    erDatoSøktSeparasjonGyldig(datoSøktSeparasjon);
+
+  return (
+    (harSøktSeparasjon?.verdi && datoSøktSeparasjonerUtfyltOgGyldig) ||
+    harSøktSeparasjon?.verdi === false
+  );
+};
+
+export const erSivilstandSpørsmålBesvart = (
+  sivilstand: string,
+  sivilstatus: ISivilstatus
+): boolean => {
+  if (erSøkerUgift(sivilstand)) {
+    return (
+      erSpørsmålOmUformeltGiftBesvart(sivilstatus) &&
+      erSpørsmålOmUformeltSeparertEllerSkiltBesvart(sivilstatus)
     );
+  } else if (erSøkerSkilt(sivilstand)) {
+    return erSpørsmålOmUformeltGiftBesvart(sivilstatus);
+  } else if (erSøkerGift(sivilstand)) {
+    return erSpørsmålOmSøktSeparasjonUtfylt(sivilstatus);
+  } else if (erSøkerEnke(sivilstand) || erSøkerSeparert(sivilstand)) {
+    return true;
+  }
+
+  return true;
+};
 
-  return ((harSøktSeparasjon?.verdi &&
-    datoSøktSeparasjonerUtfyltOgGyldig &&
-    datoFlyttetfraHverandreErUtfyltOgGyldig) ||
-    harSøktSeparasjon?.verdi === false ||
-    erSøknadsBegrunnelseBesvart(sivilstatus)) &&
+export const erStegFerdigUtfylt = (
+  sivilstatus: ISivilstatus,
+  sivilstand: string,
+  medlemskap: IMedlemskap,
+  søkerBorPåRegistrertAdresseEllerHarMeldtAdresseendring: boolean
+): boolean => {
+  return !!(
+    søkerBorPåRegistrertAdresseEllerHarMeldtAdresseendring &&
```

Comment:
Har dere testet alle permutasjonene her? :)

Reply:
Vi har testet grundig på barnetilsyn med featuretoggle slått av, men ser for oss at vi kommer til å teste litt til på overgangsstønad.